src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -268,7 +268,7 @@ public override IntermediateNode Visit(TexlCallNode node, IRTranslatorContext co
                     if (func.IsLazyEvalParam(i))
                     {
                         var child = arg.Accept(this, scope != null ? context.With(scope) : context);
-                        args.Add(new LazyEvalNode(context.GetIRContext(node), child));
```

Comment:
Why are we making this change? It means that we are treating each argument as returning the same type as the parent node.

Reply:
I think you have it reversed, that's what it was doing previously, each child had the IR Context of the call node, instead of it's own.

Reply:
Could this lead to wrong execution? Ie, Is there a .txt case that we could have added which would have failed?